tree/ntuple/v7/test/ntuple_basics.cxx

```diff
@@ -740,3 +740,22 @@ TEST(RNTuple, FillBytesWritten)
    optsSmall.SetHasSmallClusters(true);
    checkFillReturnValue(optsSmall);
 }
+
+TEST(RNTuple, RFieldDeleter)
+{
+   auto f1 = RFieldBase::Create("f1", "CustomStruct").Unwrap();
+   auto f2 = RFieldBase::Create("f2", "std::vector<std::vector<std::variant<std::string, float>>>").Unwrap();
+   auto f3 = RFieldBase::Create("f3", "std::variant<std::unique_ptr<CustomStruct>, ComplexStruct>>").Unwrap();
+   auto v1 = f1->GenerateValue();
+   auto v2 = f2->GenerateValue();
+   auto v3 = f3->GenerateValue();
+
+   // Destruct fields to check if the deleters work without the fields
+   f1 = nullptr;
+   f2 = nullptr;
+   f3 = nullptr;
+
```

Comment:
Maybe a further check in the test could be that after destructing the fields we can still call `RValue::Get` on the values and get valid objects?

Reply:
Ah maybe this would be the same irrespective of the deleter

Reply:
Yep, that's correct